### PR TITLE
Hide actions in space trash bins in case of insufficient permissions

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Actions in space trash bins are now hidden if the current user has insufficient permissions.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/7702

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/29033/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/29033/tracing/participant-management-alice-2022-10-11-12-54-43.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/29033/tracing/participant-management-anonymous-2022-10-11-12-55-03.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/29033/tracing/participant-management-brian-2022-10-11-12-54-48.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/29033/tracing/participant-management-carol-2022-10-11-12-54-52.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/mixins/actions/restore.ts [hunk 1]
```diff
@@ -29,6 +29,12 @@ export default {
               return false
             }
 
+            const { manager, editor } = this.space?.spaceRoles
+            const isProjectSpace = isProjectSpaceResource(this.space)
+            if (isProjectSpace && ![...manager, ...editor].includes(this.user.uuid)) {
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/actions/restore.ts:34]: Could you introduce two functions `isManager` and `isEditor` on the project space interface and implement them with the code you have here? That would remove some duplication... could be used as follows:
```js
if (isProjectSpaceResource(this.space) && (this.space.isManager(this.user.uuid) || this.space.isEditor(this.user.uuid))
```

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/mixins/actions/delete.js [hunk 2]
```diff
@@ -60,6 +61,16 @@ export default {
             if (this.capabilities?.files?.permanent_deletion === false) {
               return false
             }
+
+            const isProjectSpace = isProjectSpaceResource(this.space)
+            if (
+              isProjectSpace &&
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/actions/delete.js:67]:
```suggestion
              isProjectSpaceResource(this.space) &&
```
why assign it to a const? you could use it inline..

**Review by kulmann — APPROVED** [review]
nice 🤩

**Pull request merged**